### PR TITLE
CPM-801: Add mutations

### PR DESCRIPTION
**Pull request by pierallard**
- Add mutations
- Put queries in hooks
- use correct fetch mocks

### CODE REVIEW

File: components/identifier-generator/front/src/feature/hooks/useCreateIdentifierGenerator.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        const data = await response.json();
+        throw new InvalidIdentifierGenerator(data);
+      }
+
+      if (response.status !== 201) {
+        throw new ServerError();
+      }
+    },
+    {
+      onSuccess: () => queryClient.invalidateQueries('getGeneratorList'),
```
Comment (cemorin) [inline components/identifier-generator/front/src/feature/hooks/useCreateIdentifierGenerator.ts:29]: Are both of the `onSuccess` triggered? (the one from the mutate and this one)?

Comment (cemorin) [reply components/identifier-generator/front/src/feature/hooks/useCreateIdentifierGenerator.ts:29]: And you can handle it all here directly. I feel like it avoids having two places handling the response from this call (just a though/preference, not a real recommandation)

**Comment by cemorin** [comment]
:fire: :fire: :fire:

File: components/identifier-generator/front/src/feature/hooks/useCreateIdentifierGenerator.ts (hunk 1 above)
Comment (pierallard) [reply components/identifier-generator/front/src/feature/hooks/useCreateIdentifierGenerator.ts:29]: This is on purpose.
The goal of the useCreateIdentifierGenerator is to Create generator and assure that the queries are invalidated.
The goal of the onSuccess in the component is just to display notifications or redirect for end user.
Yes, the 2 onSuccess are called (you can check the doc)

**Review by Balamung — APPROVED** [review]

**Review by Beninho — APPROVED** [review]

**Pull request merged**